ravenframework/Optimizers/BayesianOptimizer.py

```diff
@@ -0,0 +1,758 @@
+    return mu, std
+
+  # * * * * * * * * * * * *
+  # Constraint Handling
+  def _applyFunctionalConstraints(self, suggested, previous):
+    """
+      Abstract method from RavenSampled, currently not used by BayesianOptimizer
+    """
+    self.raiseAnError(NotImplementedError, '_applyFunctionalConstraints not implemented for Bayesian Optimizer')
+
+  def _handleImplicitConstraints(self, previous):
+    """
+      Considers all implicit constraints
+      @ In, previous, dict, NORMALIZED previous opt point
+      @ Out, accept, bool, whether point was satisfied implicit constraints
+    """
+    normed = copy.deepcopy(previous)
+    oldVal = normed[self._objectiveVar]
+    normed.pop(self._objectiveVar, oldVal)
+    denormed = self.denormalizeData(normed)
+    denormed[self._objectiveVar] = oldVal
+    accept = self._checkImpFunctionalConstraints(denormed)
+    return accept
+
+  def _checkImpFunctionalConstraints(self, previous):
+    """
+      Checks that provided point does not violate implicit functional constraints
+      @ In, previous, dict, previous opt point (denormalized)
+      @ Out, allOkay, bool, False if violations found else True
+    """
+    allOkay = True
+    inputs = dict(previous)
+    for impConstraint in self._impConstraintFunctions:
+      okay = impConstraint.evaluate('implicitConstraint', inputs)
+      if not okay:
+        self.raiseADebug(f'Implicit constraint "{impConstraint.name}" was violated!')
+        self.raiseADebug(' ... point:', previous)
+      allOkay *= okay
+
+    return bool(allOkay)
+
+  # END constraint handling
+  # * * * * * * * * * * * *
+
+  def _resolveMultiSample(self, traj, rlz, info):
+    """
+      Handles case where multiple new datapoints are sampled.
+      Applies resolution (acceptability, improvment, constraints, rejection) to each of the points
+      @ In, traj, int, trajectory
+      @ In, rlz, xr.Datasheet, realization of multiple points
+      @ In, info, dict, information about the realization
+      @ Out, None
+    """
+    # Break out info from xr rlz into the standard realization format
+    singleRlz = {}
+    rlzVars = list(rlz.variables)
+    self.raiseADebug(f'Resolving multi-sample of size {info["batchSize"]}')
+    for index in range(info['batchSize']):
+      for varName in rlzVars:
+        singleRlz[varName] = getattr(rlz, varName)[index].values
+      optVal = singleRlz[self._objectiveVar]
+      self._resolveNewOptPoint(traj, singleRlz, optVal, info)
+      singleRlz = {} # FIXME is this necessary?
+    self.raiseADebug(f'Multi-sample resolution completed')
+
+  def _resolveNewOptPoint(self, traj, rlz, optVal, info):
+    """
+      Consider and store a new optimal point
+      @ In, traj, int, trajectory for this new point
+      @ In, info, dict, identifying information about the realization
+      @ In, rlz, xr.DataSet, batched realizations
+      @ In, optVal, list of floats, values of objective variable
+    """
```

Comment:
For this function, is it possible to leverage the same method inside RavenSampled module?

Reply:
I spent a good amount of time trying to, but to allow for the solution recommendation that the algorithm favors, the method structure inside RavenSampled does not work well.